src/main/java/model/LottoNumber.java

```diff
@@ -1,16 +1,38 @@
 package model;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
 public class LottoNumber implements Comparable<LottoNumber> {
+    public static final int LOTTO_NUMBER_LOWER_BOUND = 1;
+    public static final int LOTTO_NUMBER_UPPER_BOUND = 45;
+    private static final List<LottoNumber> CACHE = new ArrayList<>();
+
+    static {
+        for (int i = LOTTO_NUMBER_LOWER_BOUND; i <= LOTTO_NUMBER_UPPER_BOUND; i++) {
+            CACHE.add(new LottoNumber(i));
+        }
+    }
 
     private final int lottoNumber;
 
-    public LottoNumber(int lottoNumber) {
-        validateLottoNumber(lottoNumber);
-        this.lottoNumber = lottoNumber;
+    private LottoNumber(final int number) {
+        this.lottoNumber = number;
+    }
+
+    public static LottoNumber valueOf(final int number) {
+        validateLottoNumber(number);
+        LottoNumber lottoNumber = CACHE.get(number - 1);
+
+        if (Objects.isNull(lottoNumber)) {
+            lottoNumber = new LottoNumber(number);
+        }
+        return lottoNumber;
```

Comment:
CACHE.get 의 결과가 null이 나올 수 있나요? 🤔

아마도 `IndexOutOfBoundsException`이 발생하지 않을까 싶어요. 🤔
그냥 아래와 같이 변경해볼 수 있을 것 같아요. 🙂

```suggestion
        return  CACHE.get(number - 1);
```

Reply:
불필요한 코드를 제거했습니다. [d4f3907](https://github.com/next-step/java-lotto-kakao/pull/36/commits/d4f39070f69fd39b634b88f803f63a405af59857)